sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_client.py

```diff
@@ -0,0 +1,779 @@
+                self._handle_connection_close_and_no_recovery()
+            else:
+                _LOGGER.debug("WebSocket closed before open")
+                raise Exception(f"Fail to start Websocket: {close_status_code}")
+
+        if self._is_stopping:
+            raise Exception("Can't start a client during stopping")
+
+        self._ws = websocket.WebSocketApp(
+            url=url,
+            on_open=on_open,
+            on_message=on_message,
+            on_close=on_close,
+            subprotocols=[self._protocol.name] if self._protocol else [],
+        )
+
+        # set thread to start listen to server
+        self._thread = threading.Thread(target=self._ws.run_forever)
+        self._thread.start()
+        with self._cv:
+            self._cv.wait(timeout=self._start_timeout)
+        if not self._is_connected():
+            raise Exception("Fail to start client")
+
+        # set thread to check sequence id if needed
+        if self._protocol.is_reliable_sub_protocol and (
+            (self._thread_seq_ack and not self._thread_seq_ack.is_alive()) or (self._thread_seq_ack is None)
+        ):
+
+            def sequence_id_ack_periodically():
+                while self._is_connected():
+                    try:
+                        is_updated, seq_id = self._sequence_id.try_get_sequence_id()
+                        if is_updated:
+                            self._send_message(SequenceAckMessage(sequence_id=seq_id))
+                    finally:
+                        delay(1000)
+
+            self._thread_seq_ack = threading.Thread(target=sequence_id_ack_periodically)
+            self._thread_seq_ack.start()
+
+    def _start_core(self):
+        self._state = WebPubSubClientState.CONNECTING
+        _LOGGER.info("Staring a new connection")
+
+        # Reset before a pure new connection
+        self._sequence_id.reset()
+        self._is_initial_connected = False
+        self._last_close_event = None
+        self._last_disconnected_message = None
+        self._connection_id = None
+        self._reconnection_token = None
+        self._url = None
+
+        self._url = self._credential.get_client_access_url()
+        self._connect(self._url)
+
+    def start(self) -> None:
+        """start the client and connect to service"""
+
+        if self._is_stopping:
+            raise Exception("Can't start a client during stopping")
+        if self._state != WebPubSubClientState.STOPPED:
+            raise Exception("Client can be only started when it's Stopped")
+
+        try:
+            self._start_core()
+        except Exception as e:
+            self._state = WebPubSubClientState.STOPPED
+            self._is_stopping = False
+            raise e
+
+    def stop(self) -> None:
+        """stop the client"""
+
+        if self._state == WebPubSubClientState.STOPPED or self._is_stopping:
+            return
+        self._is_stopping = True
```

Comment:
Is putting `self._is_stopping = False` in `_handle_connection_stopped` a right position? If call `stop()` first, then `start()`  can't invoke `_handle_connection_stopped` and there's no chance to change back the field.

Reply:
I follow similar logic with js SDK: https://github.com/Azure/azure-sdk-for-js/blob/6a76e6ac3f67fdf7b2d22c7b3a242538b0c60f54/sdk/web-pubsub/web-pubsub-client/src/webPubSubClient.ts#L759-L763